Migrations verification fails on a Read-Only database connection

Switching to a RO database connection makes the check on the migrations to fail with the following error:
```
2023-02-10T09:05:16.391Z        ERROR   db/db.go:103    error planning migrations. Error: ERROR: cannot execute CREATE TABLE in a read-only transaction (SQLSTATE 25006)                      
/home/ubuntu/zkevm-node/log/log.go:140 github.com/0xPolygonHermez/zkevm-node/log.appendStackTraceMaybeArgs()                                                                                  
/home/ubuntu/zkevm-node/log/log.go:215 github.com/0xPolygonHermez/zkevm-node/log.Error()       
/home/ubuntu/zkevm-node/db/db.go:103 github.com/0xPolygonHermez/zkevm-node/db.checkMigrations()                                                                                               
/home/ubuntu/zkevm-node/db/db.go:54 github.com/0xPolygonHermez/zkevm-node/db.CheckMigrations()                                                                                                
/home/ubuntu/zkevm-node/cmd/run.go:148 main.checkStateMigrations()                                                                                                                            
/home/ubuntu/zkevm-node/cmd/run.go:59 main.start()                                             
/home/ubuntu/go/pkg/mod/github.com/urfave/cli/v2@v2.24.1/command.go:271 github.com/urfave/cli/v2.(*Command).Run()                                                                             
/home/ubuntu/go/pkg/mod/github.com/urfave/cli/v2@v2.24.1/command.go:264 github.com/urfave/cli/v2.(*Command).Run()                                                                             
/home/ubuntu/go/pkg/mod/github.com/urfave/cli/v2@v2.24.1/app.go:333 github.com/urfave/cli/v2.(*App).RunContext()                                                                              
/home/ubuntu/go/pkg/mod/github.com/urfave/cli/v2@v2.24.1/app.go:310 github.com/urfave/cli/v2.(*App).Run()                                                                                     
/home/ubuntu/zkevm-node/cmd/main.go:140 main.main()                                                                                                                                           
/usr/local/go/src/runtime/proc.go:250 runtime.main()                                                                                                                                          
        {"pid": 2629352, "version": "v0.0.2-RC1"}                          
github.com/0xPolygonHermez/zkevm-node/db.checkMigrations                                       
        /home/ubuntu/zkevm-node/db/db.go:103                                                   
github.com/0xPolygonHermez/zkevm-node/db.CheckMigrations                                       
        /home/ubuntu/zkevm-node/db/db.go:54                                                    
main.checkStateMigrations
        /home/ubuntu/zkevm-node/cmd/run.go:148
main.start                                                                                                                                                                                    
        /home/ubuntu/zkevm-node/cmd/run.go:59
github.com/urfave/cli/v2.(*Command).Run
        /home/ubuntu/go/pkg/mod/github.com/urfave/cli/v2@v2.24.1/command.go:271
github.com/urfave/cli/v2.(*Command).Run
        /home/ubuntu/go/pkg/mod/github.com/urfave/cli/v2@v2.24.1/command.go:264
github.com/urfave/cli/v2.(*App).RunContext
        /home/ubuntu/go/pkg/mod/github.com/urfave/cli/v2@v2.24.1/app.go:333
github.com/urfave/cli/v2.(*App).Run
        /home/ubuntu/go/pkg/mod/github.com/urfave/cli/v2@v2.24.1/app.go:310
main.main
        /home/ubuntu/zkevm-node/cmd/main.go:140 
runtime.main
        /usr/local/go/src/runtime/proc.go:250
```

--- a/db/db.go
+++ b/db/db.go
@@ -117,7 +117,9 @@ func checkMigrations(cfg Config, packrName string, direction migrate.MigrationDi
 		log.Error("error getting migrations count: ", err)
 		return err
 	}
-	if expected != actual {
+	if expected == actual {
+		log.Infof("Found %d migrations as expected", actual)
+	} else {
 		return fmt.Errorf("error the component needs to run %d migrations before starting. DB only contains %d migrations", expected, actual)
 	}
 	return nil

--- a/db/db.go
+++ b/db/db.go
@@ -97,21 +97,29 @@ func checkMigrations(cfg Config, packrName string, direction migrate.MigrationDi
 		return fmt.Errorf("packr box not found with name: %v", packrName)
 	}
 
-	var migrations = &migrate.PackrMigrationSource{Box: box}
-	planMigrations, _, err := migrate.PlanMigration(db, "postgres", migrations, direction, maxPlanMigration)
+	migrationSource := &migrate.PackrMigrationSource{Box: box}
+	migrations, err := migrationSource.FindMigrations()
 	if err != nil {
-		log.Error("error planning migrations. Error: ", err)
+		log.Errorf("error getting migrations from source: %v", err)
 		return err
 	}
-	nmigrations := len(planMigrations)
-	if nmigrations != 0 {
-		log.Errorf("error the component needs to run %d migrations before starting", nmigrations)
-		records, err := migrate.GetMigrationRecords(db, "postgres")
-		if err != nil {
-			log.Error("error getting migration records. Error: ", err)
-			return err
+
+	var expected int
+	for _, migration := range migrations {
+		if len(migration.Up) != 0 {
+			expected++
 		}
-		return fmt.Errorf("error the component needs to run %d migrations before starting. DB only contains %d migrations", nmigrations, len(records))
+	}
+
+	var actual int
+	query := `SELECT COUNT(1) FROM public.gorp_migrations`
+	err = db.QueryRow(query).Scan(&actual)
+	if err != nil {
+		log.Error("error getting migrations count: ", err)
+		return err
+	}
+	if expected != actual {
+		return fmt.Errorf("error the component needs to run %d migrations before starting. DB only contains %d migrations", expected, actual)
 	}
 	return nil
 }

--- a/db/db.go
+++ b/db/db.go
@@ -17,7 +17,6 @@ const (
 	StateMigrationName = "zkevm-state-db"
 	// PoolMigrationName is the name of the migration used by packr to pack the migration file
 	PoolMigrationName = "zkevm-pool-db"
-	maxPlanMigration  = 1000
 )
 
 var packrMigrations = map[string]*packr.Box{

